back/sssscs/src/main/java/com/ib/certificate/Certificate.java

```diff
@@ -55,11 +58,26 @@ public enum Type {
 	@Enumerated(value = EnumType.STRING)
 	private Type type;
 	
+	@Column(nullable = false)
+	@AttributeOverrides({
+		@AttributeOverride( name = "name", column = @Column(name = "subject_data_name")),
+		@AttributeOverride( name = "surname", column = @Column(name = "subject_data_surname")),
+		@AttributeOverride( name = "email", column = @Column(name = "subject_data_email")),
+		@AttributeOverride( name = "organization", column = @Column(name = "subject_data_organization")),
+		@AttributeOverride( name = "commonName", column = @Column(name = "subject_data_common_name"))
+	})
+	private SubjectData subjectData;
```

Comment:
Weird, does this work as an Embedded without the @ Embedded annotation? I thought both that one and Embeddable are required

Reply:
By the way, why did you need all these AttributeOverrides? Did something not work without them?

Reply:
I've added `@AttributeOverride` to prevent name clashing. As for `@Embedded`, it seems that sometimes it really isn't required, but omitting it was an accident. Fixed in [e851351](https://github.com/magley/sssscs/pull/5/commits/e851351bed6841fd852471d257011ef7472b84dc).

Reply:
Without attributeOverrides names clash? Wouldn't default be like: subjectdata_name and such? Like it adds the name of the field at the beginning?

Reply:
Without attributeOverrides names clash? Wouldn't default be like: subjectdata_name and such? Like it adds the name of the field at the beginning?

Reply:
Does it? I just read what it says [here](https://www.baeldung.com/jpa-embedded-embeddable#attribute-overrides) and didn't even try without the annotations.

Reply:
Ok maybe you do need those. I didn't try it out.